noah/src/main/scala/edu/uci/ics/cloudberry/noah/feed/TwitterFeedStreamDriver.java

```diff
@@ -0,0 +1,138 @@
+package edu.uci.ics.cloudberry.noah.feed;
+
+import com.google.common.collect.Lists;
+import com.twitter.hbc.ClientBuilder;
+import com.twitter.hbc.core.Client;
+import com.twitter.hbc.core.Constants;
+import com.twitter.hbc.core.endpoint.Location;
+import com.twitter.hbc.core.endpoint.StatusesFilterEndpoint;
+import com.twitter.hbc.core.processor.StringDelimitedProcessor;
+import com.twitter.hbc.httpclient.auth.Authentication;
+import com.twitter.hbc.httpclient.auth.OAuth1;
+import org.kohsuke.args4j.CmdLineException;
+import org.kohsuke.args4j.CmdLineParser;
+
+import java.io.*;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+public class TwitterFeedStreamDriver {
+
+    static Client twitterClient;
+    volatile static boolean isConnected = false;
+
+    static StreamFeedSocketAdapterClient socketAdapterClient;
+
+    public static void run(Config config)
+            throws InterruptedException, IOException {
+        BlockingQueue<String> queue = new LinkedBlockingQueue<String>(10000);
+        StatusesFilterEndpoint endpoint = new StatusesFilterEndpoint();
+        // add some track terms
+
+        if (config.getTrackTerms().length != 0) {
+            System.err.print("set track terms are: ");
+            for (String term : config.getTrackTerms()) {
+                System.err.print(term);
+                System.err.print(" ");
+            }
+            System.err.println();
+            endpoint.trackTerms(Lists.newArrayList(config.getTrackTerms()));
+        }
+        if (config.getTrackLocation().length != 0) {
+
+            System.err.print("set track locations are:");
+            for (Location location : config.getTrackLocation()) {
+                System.err.print(location);
+                System.err.print(" ");
+            }
+            System.err.println();
+
+            endpoint.locations(Lists.<Location>newArrayList(config.getTrackLocation()));
+        }
+
+        Authentication auth = new OAuth1(config.getConsumerKey(), config.getConsumerSecret(), config.getToken(),
+                config.getTokenSecret());
+
+        // Create a new BasicClient. By default gzip is enabled.
+        twitterClient = new ClientBuilder()
+                .hosts(Constants.STREAM_HOST)
+                .endpoint(endpoint)
+                .authentication(auth)
+                .processor(new StringDelimitedProcessor(queue))
+                .build();
+
+        SimpleDateFormat sdfDate = new SimpleDateFormat("yyyy-MM-dd_HH:mm:ss");
```

Comment:
it's better to replace `:` with some the other marker if it is part of filename.